packages/staking/src/features/store/delegationPortfolioStore/useDelegationPortfolioStore.ts

```diff
@@ -109,53 +87,69 @@ export const useDelegationPortfolioStore = create(
-          decimals: 0,
-          items: delegationDistribution.map((item) => ({
-            ...item,
-            percentage: item.percentage * PERCENTAGE_SCALE_MAX,
-          })),
-          key: 'percentage',
-        })
+        let mapOfPoolIdBySavedPercentages: Record<Wallet.Cardano.PoolIdHex, number> = {};
+        if (delegationPortfolio?.pools) {
+          const weightsSum = delegationPortfolio.pools.reduce((acc, { weight }) => acc + weight, 0);
+          const portfolioWithPercentages = delegationPortfolio.pools.map((pool) => ({
+            ...pool,
+            percentage: (pool.weight / weightsSum) * PERCENTAGE_SCALE_MAX,
+          }));
           // eslint-disable-next-line unicorn/no-array-reduce
-          .reduce((acc, item) => {
-            acc[item.pool.hexId] = item.percentage;
-            return acc;
-          }, {} as Record<Wallet.Cardano.PoolIdHex, number>);
+          mapOfPoolIdBySavedPercentages = portfolioWithPercentages.reduce<Record<Wallet.Cardano.PoolIdHex, number>>(
+            (acc, { percentage, id }) => {
+              acc[id] = percentage;
+              return acc;
+            },
+            {}
+          );
+        }
 
-        const currentPortfolio = delegationDistribution.map(({ pool: stakePool, percentage, stake }) => {
-          const confirmedPoolRewards = confirmedRewardHistory
-            .filter(({ poolId }) => poolId === stakePool.id)
-            .map(({ rewards }) => rewards);
+        let currentPortfolio = delegationDistribution.map<CurrentPortfolioStakePool>(
+          ({ pool: stakePool, percentage, stake }) => {
+            const confirmedPoolRewards = confirmedRewardHistory
+              .filter(({ poolId }) => poolId === stakePool.id)
+              .map(({ rewards }) => rewards);
 
-          return {
-            displayData: {
-              ...mapStakePoolToDisplayData({ cardanoCoinSymbol: get().cardanoCoinSymbol, stakePool }),
-              lastReward: confirmedPoolRewards[confirmedPoolRewards.length - 1] || BigInt(0),
-              totalRewards: Wallet.BigIntMath.sum(confirmedPoolRewards),
-            },
-            id: stakePool.hexId,
-            onChainPercentage: percentage * PERCENTAGE_SCALE_MAX,
-            savedIntegerPercentage: savedPercentages[stakePool.hexId] || 0,
-            sliderIntegerPercentage: savedPercentages[stakePool.hexId],
-            stakePool,
-            value: stake,
-          };
+            return {
+              displayData: {
+                ...mapStakePoolToDisplayData({ cardanoCoinSymbol: get().cardanoCoinSymbol, stakePool }),
+                lastReward: confirmedPoolRewards[confirmedPoolRewards.length - 1] || BigInt(0),
+                totalRewards: Wallet.BigIntMath.sum(confirmedPoolRewards),
+              },
+              id: stakePool.hexId,
+              onChainPercentage: percentage * PERCENTAGE_SCALE_MAX,
+              savedIntegerPercentage: mapOfPoolIdBySavedPercentages[stakePool.hexId] || 0,
```

Comment:
I think it would be better for readability if we explicitly captured with `null` that the given saved percentage is just missing:

```suggestion
              savedIntegerPercentage: mapOfPoolIdBySavedPercentages[stakePool.hexId] ?? null,
```

e.g. `isPortfolioDrifted()` helper and the check for missing percentage would become more readable as checking for `zero` (which one may argue is a valid value) is IMHO unnecessarily "hacky"

Reply:
I considered `null` at first, but I realized it requires more changes in places, where we read the value. In those places, we would need to fall back it to 0 anyway. Not in all the places, though. I am tentative about this, but I support your argument about readability.